test/e2e/sample_e2e_test.go

```diff
@@ -0,0 +1,174 @@
+			{Version: semver.MustParse("1.0.0"), ReplacesVersion: "", SkipRange: "", DefaultChannel: alphaChannel, Channels: []string{alphaChannel}, NewIndex: false, PackageName: "testoperatorb", OwnedGVKs: cu.B1v1CRDDescription, DependencyGVKs: cu.A1v1CRDDescription, DependencyPackages: nil, Addmode: cu.SemverSkipPatch, ConfigMap: nil, Secret: nil, CrdVersions: cu.V1CRDVersionV1beta1, OperatorImage: operatorImage, OperatorCommand: operatorCommand},
+		}
+		// TEST FOR DOWNSTREAM REGISTRY
+
+		// stack :=  cu.Stack{
+		// 	OpmBinarySourceImage: cu.Downstream4_6,
+		// 	CatalogFromImage:     cu.Ubi8,
+		// 	CatalogName:          "panamax",
+		// 	CatalogTag:           "latest",
+		// 	Oc:                   cu.Ocv4_5_0,
+		// 	Opmdown:           cu.Opmdownv1_14_3,
+		// 	OpmDebug:          true,
+		// 	TargetRegistry:    "localhost:5000",
+		// 	ContainerCLI:      cu.Docker,
+		// 	TargetCatalogType: cu.Registry,
+		// }
+
+		// TEST FOR UPSTREAM REGISTRY
+
+		// stack :=  cu.Stack{
+		// 	OpmBinarySourceImage: cu.Upstream1_15,
+		// 	CatalogFromImage:     cu.Ubi8,
+		// 	CatalogName:          "panamax1",
+		// 	CatalogTag:           "latest",
+		// 	Oc:                   cu.Ocv4_5_0,
+		// 	// Opmdown:              cu.Opmdownv1_14_3,
+		// 	Opmup:             cu.Opmupv1_15_2,
+		// 	OpmDebug:          true,
+		// 	TargetRegistry:    "localhost:5000",
+		// 	ContainerCLI:      Docker,
+		// 	TargetCatalogType: Registry,
+		// }
+
+		// TEST FOR DOWNSTREAM IMAGE --- NOTE THIS DOES NOT WORK
+		// Hit Error: error copying container directory open /private/var/folders/f_/8tjmq3h93jq1yk1z4k3zrx0w0000gn/T/catalog769363132/index_tmp_766319566/root/.bash_logout: permission denied
+
+		// stack :=  cu.Stack{
+		// 	OpmBinarySourceImage: cu.Downstream4_6,
+		// 	CatalogFromImage:     cu.Ubi8,
+		// 	CatalogName:          "panamax2",
+		// 	CatalogTag:           "latest",
+		// 	Oc:                   cu.Ocv4_5_0,
+		// 	Opmdown:              cu.Opmdownv1_14_3,
+		// 	OpmDebug:             true,
+		// 	TargetRegistry:       "localhost:5000",
+		// 	ContainerCLI:         cu.Docker,
+		// 	TargetCatalogType:    cu.Image,
+		// }
+
+		// TEST FOR UPSTREAM IMAGE
+		stack := cu.Stack{
+			OpmBinarySourceImage: cu.Upstream1_15,
+			CatalogFromImage:     cu.Ubi8,
+			CatalogName:          "panamax3",
+			CatalogTag:           "latest",
+			Oc:                   cu.Ocv4_5_0,
+			// Opmdown:              cu.Opmdownv1_14_3,
+			Opmup:             cu.Opmupv1_15_2,
+			OpmDebug:          true,
+			TargetRegistry:    "localhost:5000",
+			ContainerCLI:      cu.Docker,
+			TargetCatalogType: cu.Image,
+		}
+
+		err := cu.CreateTemporaryCatalog(*toolsBin, catalogEntry, stack)
+		_ = err
+	})
+
+	Describe("test with catalog index image created in replaces mode", func() {
+
+		It("create subscription with no dependency", func() {
```

Comment:
I know this is just an example test, but I just want to point out that we do accomplish the same functionality by testing each discrete unit instead. For example, given an installplan, do we get the expected objects on cluster. From a subscription to a catalog source, do we get the expected resolution and an installplan with the right items in it on-cluster. 

These overarching tests can be extremely powerful if we come up with some scenarios that are truly beyond what testing each individual piece does. Most of the e2e test cases that we write as part of bug-fixing are extremely specific and tied to edge cases in the individual APIs versus the overall interaction of the system. The overall interaction is interesting, we should brainstorm some interesting scenarios that we can test.

Reply:
I know that @cdjohnson has a whole set of test plans written down that we were going to implement.